src/java/org/apache/cassandra/cql3/selection/Selector.java

```diff
@@ -443,14 +447,21 @@ private <V> ByteBuffer value(Cell<V> c)
         }
 
         /**
-         * Return the value of the column with the specified index.
+         * Return the value of the column with the specified index. If the column the value belongs to is masked with a
+         * {@link ColumnMask} and {@link #unmask} hasn't been specified, such mask will be applied to the value.
          *
          * @param index the column index
-         * @return the value of the column with the specified index
+         * @return the value of the column with the specified index, masked if its column is masked
          */
         public ByteBuffer getValue(int index)
         {
-            return values[index];
+            ByteBuffer value = values[index];
```

Comment:
I looked into the code and my confusion was coming from the fact that `SimpleSelection` relies on `InputRow.getValues()`  (https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/cql3/selection/Selection.java#L478) and not `getValue`. Nevertheless, it introduce some weird behavior has one method will return masked values and not the other one. I think that we should be consistent. We could either do those operations at the `InputRow` level and do it when the value is set (which would allow us to simplify the logic at the SelectStatement level) or keep the logic in `SimpleSelector `.
What is also unclear to me is how do we plan to handle the post ordering (https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/cql3/statements/SelectStatement.java#L1059)?

Reply:
Forbidding `ORDER BY` can be a pain as it would require changing the CQL in your app to toggle behaviors. If they're concerned on ordering they can do a partial hash, use a bespoke masking function,... they're not perfect solutions but they sound better than the added cost to sort by masked values imo.